src/compile/invalid/ScaleInvalidDataMode.ts

```diff
@@ -0,0 +1,51 @@
+import {SignalRef} from 'vega';
+import {ScaleChannel} from '../../channel';
+import {Config} from '../../config';
+import {MarkInvalidDataMode} from '../../invalid';
+import {MarkDef, isPathMark} from '../../mark';
+import {ScaleType, hasContinuousDomain} from '../../scale';
+import {getMarkPropOrConfig} from '../common';
+import {normalizeInvalidDataMode} from './normalizeInvalidDataMode';
+
+export type ScaleInvalidDataMode =
+  // remove 'break-paths-and-keep-path-domains' from MarkInvalidDataMode
+  // because it is a macro for '"filter"' or `"break-path-keep-domains`
+  | Omit<MarkInvalidDataMode, 'break-paths-and-keep-path-domains'>
+
+  // Add always-valid because at scale level, categorical scales can handle any values and thus is always valid.
+  | 'always-valid';
```

Comment:
what if we have `null` and "null" in data? `null` is as "valid" as "null"`. Maybe we want to allow the users to customize null text later.

Reply:
Yeah, I agree it's an issue, but it's a separate issue orthogonal to this PR (which focuses on continuous scales).

Reply:
I didn't find one searching `null categorical `. I'll defer to when we need that in Helios, since I guess you'll be busy and won't have the chance to add it if it's not requested from Helios.